Lib/shlex.py

```diff
@@ -312,26 +316,78 @@ def split(s, comments=False, posix=True):
     lex.whitespace_split = True
     if not comments:
         lex.commenters = ''
-    return list(lex)
+
+    if isinstance(s, bytes):
+        return [i.encode("ascii") for i in lex]
+    else:
+        return list(lex)
 
 
 def join(split_command):
     """Return a shell-escaped string from *split_command*."""
-    return ' '.join(quote(arg) for arg in split_command)
+    if len(split_command) == 0:
```

Comment:
This behaviour should be documented, saying that a string will be returned instead of bytes for an empty sequence of bytes. Or maybe add an additional a separate function which only accepts bytes inputs (less code and no warnings and more efficient).